### ISSUE TITLE
Avoid memset with NULL ptr for 0 bytes

### ISSUE THREAD

**Issue by jeroenooms** [wesm/feather#170]
One of the problems as raised by ASAN in https://github.com/wesm/feather/issues/163 for data frames appears for data frames with 0 rows. 

The problem is that **`memset()` may never be called `ptr = NULL` even when `num == 0`**. This problem appears several times in [feather-write.cpp](https://github.com/wesm/feather/blob/master/R/src/feather-write.cpp#L11-L19) when `nbytes = 0` for example:

```cpp
auto buffer = std::make_shared<OwnedMutableBuffer>();
stopOnFailure(buffer->Resize(nbytes));
memset(buffer->mutable_data(), 0, nbytes);
```

The same problem appears later again when `AppendPrimitive` gets called with a null pointer for the `&values` parameter, which is not allowed.  

There are two possible solutions:

 1. condition each call to `memset` on `if(nbytes > 0)` and change `&values` to `*values`
 2. make `OwnedMutableBuffer` return a dummy pointer for `buffer->Resize(0)`

I think the latter would be more correct.

**Comment by wesm** [comment]
I'm in favor if solution 1

**Comment by wesm** [comment]
I believe this was addressed by #207 / #210. Closing
